### ISSUE TITLE
Don't rely on heavy tests for simple things

### ISSUE THREAD

**Issue by adriancole** [spring-cloud/spring-cloud-sleuth#100]
I don't think that we need rabbit to do normal tests. Ex. the "integration" profile is quite heavy as it relies on docker and take a long time to start and is annoying to debug as the console log is cluttered with all sorts of stuff. Moreover, the kind of problems caught are often just minor code issues. These should be caught in a way that run normally.

For example, property tests around input spans being received by http endpoint shouldn't in any way need to engage rabbit to assert. Ex. http 400 comes way before any of that matters.

Long story short, is lets try to keep our normal path tested without heavy dependencies such as rabbit or docker. I'd be surprised if spring-cloud-stream has no in-memory alternative, for example.

Let's only use docker as a last resort as debugging it makes for a very poor experience in troubleshooting, even if easier and faster than the brewery tests.

**Comment by adriancole** [comment]
@dsyer quick one.. I'm assuming we can test spring-cloud-stream basic functionality in a unit test w/o using docker. Is that a valid assumption?

**Comment by adriancole** [comment]
PS a common way I've seen this achieved is that a base class includes the spec that is easiest to run. A sub-class or other configuration affects the test such that the same thing passes with different dependencies. Ex. explicitly we could have a subclass that engages docker. Ex. implicitly (like arquillian) a configuration swaps the backend in a profile)

**Comment by adriancole** [comment]
PS let's remember zipkin-java-server is a boot/mvc app. Even if test runners don't support multiple containers (like arquillian does), it may be possible to just host the query api in a test server. That could lead to a quite fast path for debugging issues like span properties being null and dropped accordingly.

**Comment by dsyer** [comment]
Yes, there's not much point straining with docker if everything you need runs in the JVM.

**Comment by marcingrzejszczak** [comment]
Done - https://github.com/spring-cloud/spring-cloud-sleuth/commit/4efa4c3f27cc74ad115134554c5c3d476191361f
